deploy/helm/sumologic/templates/metrics/otelcol/statefulset.yaml

```diff
@@ -0,0 +1,139 @@
+                values:
+                - {{ template "sumologic.labels.app.logs" . }}
+                - {{ template "sumologic.labels.app.metrics" . }}
+              - key: app
+                operator: In
+                values:
+                - prometheus-operator-prometheus
+            topologyKey: "kubernetes.io/hostname"
+{{else if eq .Values.metrics.statefulset.podAntiAffinity "soft"}}
+        podAntiAffinity:
+          preferredDuringSchedulingIgnoredDuringExecution:
+          - weight: 100
+            podAffinityTerm:
+              labelSelector:
+                matchExpressions:
+                - key: app
+                  operator: In
+                  values:
+                  - {{ template "sumologic.labels.app.logs.pod" . }}
+                  - {{ template "sumologic.labels.app.metrics.pod" . }}
+                - key: app
+                  operator: In
+                  values:
+                  - prometheus-operator-prometheus
+              topologyKey: "kubernetes.io/hostname"
+{{- end }}
+{{- end }}
+      volumes:
+      - name: config-volume
+        configMap:
+          name: {{ template "sumologic.metadata.name.metrics.configmap" . }}
+{{- if .Values.metrics.extraVolumes }}
+{{ toYaml .Values.metrics.extraVolumes | indent 6 }}
+{{- end }}
+      securityContext:
+        {{- toYaml .Values.otelcol.securityContext | nindent 8 }}
+      {{- if .Values.metrics.statefulset.priorityClassName }}
+      priorityClassName: {{ .Values.metrics.statefulset.priorityClassName | quote }}
+      {{- end }}
+      containers:
+      - name: otelcol
+        image: {{ .Values.otelcol.statefulset.image.repository }}:{{ .Values.otelcol.statefulset.image.tag }}
+        imagePullPolicy: {{ .Values.otelcol.statefulset.image.pullPolicy }}
+        resources:
+          {{- toYaml .Values.metrics.statefulset.resources | nindent 10 }}
+        {{- if .Values.metrics.statefulset.containers.otelcol.securityContext }}
+        securityContext:
+          {{- toYaml .Values.metrics.statefulset.containers.otelcol.securityContext | nindent 10 }}
+        {{- end }}
+        ports:
+        - name: prom-write
+          containerPort: 9888
+          protocol: TCP
+        livenessProbe:
+          httpGet:
+            path: /
+            port: 13133 # Health Check extension default port.
+        readinessProbe:
+          httpGet:
+            path: /
+            port: 13133 # Health Check extension default port.
+        volumeMounts:
+        - name: config-volume
+          mountPath: /etc/otel/config.yaml
+          subPath: config.yaml
+{{- if .Values.metrics.extraVolumeMounts }}
+{{ toYaml .Values.metrics.extraVolumeMounts | indent 8 }}
+{{- end }}
+        env:
+{{- $ctx := .Values -}}
+{{ include "kubernetes.sources.envs" (dict "Context" $ctx "Type" "metrics")}}
+        {{- if .Values.metrics.extraEnvVars }}
+{{ toYaml .Values.metrics.extraEnvVars | nindent 8 }}
+        {{- end }}
+{{- end }}
```

Comment:
It may be useful to add option to set` args` for otel-col to change for example `log-level`.

Reply:
done :)